crates/oxc_linter/src/rules/unicorn/prefer_structured_clone.rs

```diff
@@ -0,0 +1,206 @@
+                inner_call_expr,
+                Some(&["JSON"]),
+                Some(&["stringify"]),
+                Some(1),
+                Some(1),
+            ) {
+                return;
+            }
+
+            if inner_call_expr.arguments[0].is_spread() {
+                return;
+            }
+
+            let span = Span::new(call_expr.span.start, inner_call_expr.span.end);
+            ctx.diagnostic(prefer_structured_clone_diagnostic(span));
+        } else if !call_expr.arguments[0].is_spread() {
+            for function in &self.allowed_functions {
+                if let Some((object, method)) = function.split_once('.') {
+                    if is_method_call(call_expr, Some(&[object]), Some(&[method]), None, None) {
+                        ctx.diagnostic(prefer_structured_clone_diagnostic(call_expr.span));
+                    }
+                } else if is_method_call(call_expr, None, Some(&[function]), None, None)
+                    || is_method_call(call_expr, Some(&[function]), None, None, None)
+                    || is_specific_function_call(call_expr, function)
+                {
+                    ctx.diagnostic(prefer_structured_clone_diagnostic(call_expr.span));
+                }
+            }
+        }
+    }
+}
+
+fn is_specific_function_call(call_expr: &CallExpression, function_name: &String) -> bool {
+    if let Expression::Identifier(ident) = &call_expr.callee {
+        return ident.name == function_name;
+    }
+
+    false
+}
+
+#[test]
+fn test() {
+    use crate::tester::Tester;
+
+    let pass = vec![
+        ("structuredClone(foo)", None),
+        ("JSON.parse(new JSON.stringify(foo))", None),
+        ("new JSON.parse(JSON.stringify(foo))", None),
+        ("JSON.parse(JSON.stringify())", None),
+        ("JSON.parse(JSON.stringify(...foo))", None),
+        ("JSON.parse(JSON.stringify(foo, extraArgument))", None),
+        ("JSON.parse(...JSON.stringify(foo))", None),
+        ("JSON.parse(JSON.stringify(foo), extraArgument)", None),
+        ("JSON.parse(JSON.stringify?.(foo))", None),
+        ("JSON.parse(JSON?.stringify(foo))", None),
+        ("JSON.parse?.(JSON.stringify(foo))", None),
+        // ("JSON?.parse(JSON.stringify(foo))", None),
+        ("JSON.parse(JSON.not_stringify(foo))", None),
+        ("JSON.parse(not_JSON.stringify(foo))", None),
+        ("JSON.not_parse(JSON.stringify(foo))", None),
+        ("not_JSON.parse(JSON.stringify(foo))", None),
+        ("JSON.stringify(JSON.parse(foo))", None),
+        ("JSON.parse(JSON.stringify(foo, undefined, 2))", None),
+        ("new _.cloneDeep(foo)", None),
+        ("notMatchedFunction(foo)", None),
+        ("_.cloneDeep()", None),
+        ("_.cloneDeep(...foo)", None),
+        ("_.cloneDeep(foo, extraArgument)", None),
+        // ("_.cloneDeep?.(foo)", None),
+        // ("_?.cloneDeep(foo)", None),
+    ];
+
+    let fail = vec![
+        ("JSON.parse(JSON.stringify(foo))", None),
```

Comment:
can you add a test for `JSON.parse((JSON.stringify((foo))))` please

Reply:
Done and that fails:

```
---- rules::unicorn::prefer_structured_clone::test stdout ----
thread 'rules::unicorn::prefer_structured_clone::test' panicked at crates/oxc_linter/src/tester.rs:322:13:
expect test to fail: JSON.parse((JSON.stringify((foo))))
note: run with `RUST_BACKTRACE=1` environment variable to display a backtrace
```

Reply:
i think you might need a `without_parenthesis` on L92?

Reply:
So this needs `without_parenthesized()`

Reply:
yeah i think L92, where you checkl the args should be:
```rust
let Argument::CallExpression(inner_call_expr) = &call_expr.arguments[0].without_parenthesis() else {

Reply:
ah actually might need to be

```rust
        let Some(Expression::CallExpression(inner_call_expr)) =
            self.arguments[0].as_expression().map(|v| v.without_parenthesized())
        else {
            return;
        };
 ```